src/test/regress/upgrade/config.py

```diff
@@ -1,68 +1,298 @@
+            self.node_name_to_ports[cur_worker_name] = cur_port
+            self.worker_ports.append(cur_port)
+
+    def _get_and_update_next_port(self):
+        if hasattr(self, "fixed_port"):
+            next_port = self.fixed_port
+            self.fixed_port += 1
+            return next_port
+        return find_free_port()
+
+
+class CitusDefaultClusterConfig(CitusBaseClusterConfig):
     def __init__(self, arguments):
-        self.old_bindir = arguments['--old-bindir']
-        self.new_bindir = arguments['--new-bindir']
-        self.pg_srcdir = arguments['--pgxsdir']
-        self.temp_dir = './tmp_upgrade'
-        self.old_datadir = self.temp_dir + '/oldData'
-        self.new_datadir = self.temp_dir + '/newData'
-        self.settings = {
-            'shared_preload_libraries': 'citus',
-            'citus.node_conninfo': 'sslmode=prefer'
+        super().__init__(arguments)
+        new_settings = {
+            "client_min_messages": "WARNING",
         }
+        self.settings.update(new_settings)
 
 
-USER = 'postgres'
-DBNAME = 'postgres'
+class CitusMXBaseClusterConfig(CitusDefaultClusterConfig):
+    def __init__(self, arguments):
+        super().__init__(arguments)
+        self.is_mx = True
+        self.add_coordinator_to_metadata = True
+
 
-COORDINATOR_NAME = 'coordinator'
-WORKER1 = 'worker1'
-WORKER2 = 'worker2'
-NODE_NAMES = [COORDINATOR_NAME, WORKER1, WORKER2]
-COORDINATOR_PORT = 57635
-WORKER1PORT = 57636
-WORKER2PORT = 57637
+class CitusUpgradeConfig(CitusBaseClusterConfig):
+    def __init__(self, arguments):
+        super().__init__(arguments)
+        self.pre_tar_path = arguments["--citus-pre-tar"]
+        self.post_tar_path = arguments["--citus-post-tar"]
+        self.temp_dir = "./tmp_citus_upgrade"
+        self.new_settings = {"citus.enable_version_checks": "false"}
+        self.user = SUPER_USER_NAME
+        self.mixed_mode = arguments["--mixed"]
+        self.fixed_port = 57635
 
-WORKER_PORTS = [WORKER1PORT, WORKER2PORT]
-NODE_PORTS = {
-    COORDINATOR_NAME: COORDINATOR_PORT,
-    WORKER1: WORKER1PORT,
-    WORKER2: WORKER2PORT,
-}
+
+class PostgresConfig(CitusDefaultClusterConfig):
+    def __init__(self, arguments):
+        super().__init__(arguments)
+        self.worker_amount = 0
+        self.is_citus = False
+
+
+class CitusSingleNodeClusterConfig(CitusDefaultClusterConfig):
+    def __init__(self, arguments):
+        super().__init__(arguments)
+        self.worker_amount = 0
+
+
+class CitusSingleWorkerClusterConfig(CitusMXBaseClusterConfig):
+    def __init__(self, arguments):
+        super().__init__(arguments)
+        self.worker_amount = 1
+
+
+class CitusSuperUserDefaultClusterConfig(CitusMXBaseClusterConfig):
+    def __init__(self, arguments):
+        super().__init__(arguments)
+        self.user = SUPER_USER_NAME
+
+
+class CitusThreeWorkersManyShardsClusterConfig(CitusMXBaseClusterConfig):
+    def __init__(self, arguments):
```

Comment:
Lets increase "max_connections": 300 to something higher here. Again, because of repartitions, there are way too many connections are created :/  So, 600 should be good for now

Reply:
sure done